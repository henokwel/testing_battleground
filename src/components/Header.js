import React from 'react'

export const Header = () => {
    return (
        <header className="App-header">
            <h1>Header</h1>
        </header>
    )
}
