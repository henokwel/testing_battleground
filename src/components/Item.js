import React, { useState } from 'react'
import { useMyContext } from '../context/MyContext'

export const Item = ({ id, title, done }) => {
    const { dispatch } = useMyContext()
    const [edit, setEdit] = useState(false)
    const [currentTitle, setCurrentTitle] = useState(title)
    const [currentDone, setCurrentDone] = useState(done)

    const handleChange = (e) => {
        const target = e.target
        const value = target.value
        console.log(value);
        // if (!value || value === title) return
        setCurrentTitle(value)
    }

    const handleRemoveItem = (id) => {
        dispatch({ type: "remove", payload: id })
    }

    const handleEdit_Title = (id) => {
        // if current is false, toggle
        if (!edit) {
            setEdit(!edit)
        } else {
            // if current is true, compare and disptach
            if (currentTitle === title || currentTitle.trim() === "") return setEdit(false)
            dispatch({ type: "edit", payload: { id, title: currentTitle } })
            setEdit(!edit)
        }
    }

    const handleEdit_Done = (id) => {
        // if current is false, toggle
        dispatch({ type: "done", payload: { id, done: !currentDone } })
        setCurrentDone(!currentDone)
    }

    // onSubmit={(e) => e.preventDefault()}
    return (
        <div className='item'  >            
            {
                !edit ?
                    <p>{currentTitle}</p>
                    :
                    <input type="text" value={currentTitle} onChange={(e) => handleChange(e)} />
            }

            <div className='item_btn_Container'>
                <input type="checkbox" checked={currentDone} onChange={() => handleEdit_Done(id)} />
                <button onClick={() => handleEdit_Title(id)}>{edit ? "Done" : "Edit"}</button>
                <button onClick={() => handleRemoveItem(id)}>Remove</button>
            </div>

        </div>
    )
}
