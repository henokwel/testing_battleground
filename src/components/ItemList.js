import React, { useContext } from 'react'
import { Item } from '../components/Item'
import { Context } from '../context/MyContext'



export const ItemList = () => {
    const { state } = useContext(Context)


    return (
        <div className='itemList' >

            {
                !state ? <p>No Data</p> :
                    state.map((item, index) => <Item key={index}  {...item} />)
            }
        </div>
    )
}
