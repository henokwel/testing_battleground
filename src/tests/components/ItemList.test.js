import { shallow } from "enzyme";
import { ItemList } from "../../components/ItemList";
import { data } from "../fixtures/data"

describe('ItemList', () => {

    test('should render itemList with no data', () => {
        const wrapper = shallow(<ItemList data={[]} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should render itemlist with', () => {
        const wrapper = shallow(<ItemList data={data} />)
        expect(wrapper).toMatchSnapshot()
    })
    


})
