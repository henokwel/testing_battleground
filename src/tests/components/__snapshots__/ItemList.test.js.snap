// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList should render itemList with no data 1`] = `
<div
  className="itemList"
>
  <p>
    No Data
  </p>
</div>
`;

exports[`ItemList should render itemlist with 1`] = `
<div
  className="itemList"
>
  <Item
    done={false}
    id={1}
    key="1"
    title="One"
  />
  <Item
    done={false}
    id={2}
    key="2"
    title="two"
  />
  <Item
    done={false}
    id={3}
    key="3"
    title="three"
  />
</div>
`;
